kanidm_book/src/examples/k8s_ingress_example.md

```diff
@@ -0,0 +1,278 @@
+      k8s-app: oauth2-proxy-kanidm-example
+  template:
+    metadata:
+      labels:
+        k8s-app: oauth2-proxy-kanidm-example
+    spec:
+      containers:
+      - args:
+        - --provider=oidc
+        - --email-domain=*
+        - --upstream=file:///dev/null
+        - --http-address=0.0.0.0:4182
+        - --oidc-issuer-url=https://<kosrim-domain>/oauth2/openid/<oauth2_rs_name>
+        - --code-challenge-method=S256
+        env:
+        - name: OAUTH2_PROXY_CLIENT_ID
+          value: <oauth2_rs_name>
+        - name: OAUTH2_PROXY_CLIENT_SECRET
+          value: <oauth2_rs_basic_secret>
+        - name: OAUTH2_PROXY_COOKIE_SECRET
+          value: <cookie_secret> # output b`cookie_secret` from docker run -ti --rm python:3-alpine python -c 'import secrets,base64; print(base64.b64encode(base64.b64encode(secrets.token_bytes(16))));'
+        image: quay.io/oauth2-proxy/oauth2-proxy:latest
+        imagePullPolicy: Always
+        name: oauth2-proxy-kanidm-example
+        ports:
+        - containerPort: 4182
+          protocol: TCP
+        securityContext:
+          allowPrivilegeEscalation: false
+          capabilities:
+            drop: ["ALL"]
+      securityContext:
+        runAsNonRoot: true
+        seccompProfile:
+          type: RuntimeDefault
+---
+apiVersion: v1
+kind: Service
+metadata:
+  labels:
+    k8s-app: oauth2-proxy-kanidm-example
+  name: oauth2-proxy-kanidm-example
+  namespace: kanidm-example
+spec:
+  ports:
+  - name: http
+    port: 4182
+    protocol: TCP
+    targetPort: 4182
+  selector:
+    k8s-app: oauth2-proxy-kanidm-example
+
+---
+apiVersion: networking.k8s.io/v1
+kind: Ingress
+metadata:
+  name: oauth2-proxy-kanidm-example
+  namespace: kanidm-example
+spec:
+  ingressClassName: nginx
+  rules:
+  - host: <hostname>
+    http:
+      paths:
+      - path: /oauth2
+        pathType: Prefix
+        backend:
+          service:
+            name: oauth2-proxy-kanidm-example
+            port:
+              number: 4182
+  tls:
+  - hosts:
+    - <hostname>
+    secretName: <hostname>-ingress-tls # replace . with - in the hostname
+```
+
+3. Apply the configuration by running the following command:
+
+```shell
+kubectl apply -f k8s.kanidm-nginx-auth-example.yaml
+```
+
+4. Check your deployment succeeded by creating `check.sh` with the block below and running:
+```shell
+#!/bin/sh
+
+# Analysis self for common errors.
+shellcheck ./*.sh || exit 1
+
+kubectl -n kanidm-example get all
+kubectl -n kanidm-example get ingress
+kubectl -n kanidm-example get Certificate
+
+# kubctl -n kanidm-example describe and kubctl -n kanidm-example logs will do you well for troubleshooting.
+# If there are ingress errors: https://kubernetes.github.io/ingress-nginx/troubleshooting/
+# If there are certificate errors: https://cert-manager.io/docs/faq/troubleshooting/
+```
+
+Once it has finished deploying, you should be able to access it at `https://<hostname>`, which will prompt you for authentication.
```

Comment:
Is it on port 443 or 8080? I can't read k8syaml yet 😄

Reply:
443 :)